src/System.Collections.NonGeneric/tests/Hashtable/PropertyItemThreadSafety.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.Collections;
+using System.Threading;
+using System.Threading.Tasks;
+
+/// <summary>
+/// the hashtable has a race:
+///     A read operation on hashtable has three steps:
+///        (1) calculate the hash and find the slot number.
+///        (2) compare the hashcode, if equal, go to step 3. Otherwise end.
+///        (3) compare the key, if equal, go to step 4. Otherwise end.
+///        (4) return the value contained in the bucket.
+///     The problem is that after step 3 and before step 4. A writer can kick in a remove the old item and add a new one 
+///     in the same bukcet. In order to make this happen easily, I created two long with same hashcode.
+/// </summary>
+public class ItemThreadSafetyTests
+{
+    private object _key1;
+    private object _key2;
+    private object _value1 = "value1";
+    private object _value2 = "value2";
+    private Hashtable _ht;
+
+    private bool _errorOccured = false;
+    private bool _timeExpired = false;
+
+    private const long MAX_TEST_TIME_TICKS = 10L * 10000000L; // 10 seconds
+
+    [Fact]
+    [OuterLoop]
+    public void TestGetItemThreadSafety()
+    {
+        int i1 = 0x10;
+        int i2 = 0x100;
+        long l1;
+        long l2;
+
+        //Setup key1 and key2 so they are different values but have the same hashcode
+        //To produce a hashcode long XOR's the first 32bits with the last 32 bits
+        l1 = (((long)i1) << 32) + i2;
+        l2 = (((long)i2) << 32) + i1;
+        _key1 = (object)l1;
```

Comment:
Same as a previous comment: The functionality of this test is depending on long.GetHashCode's implementation. An assert to codify that dependency would ensure the test is kept up-to-date.